File: Arrays/find-all-numbers-disappeared-in-an-array/find_disappeared_numbers.py


from typing import List


def find_disappeared_numbers(nums: List[int]) -> List[int]:

    return [i for i in range(1, len(nums)+1) if i not in nums]


if __name__ == '__main__':

    print(find_disappeared_numbers([4, 3, 2, 7, 8, 2, 3, 1]))
    print(find_disappeared_numbers([4, 3, 2, 7, 7, 2, 3, 1]))
    print(find_disappeared_numbers([1, 1]))

File: Arrays/find-all-numbers-disappeared-in-an-array/find_disappeared_numbers.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> find_disappeared_numbers(vector<int> &nums)
{
    unordered_set<int> set_nums(begin(nums), end(nums));
    vector<int> res;

    for (int i = 1; i <= nums.size(); i++)
    {
        if (!set_nums.count(i))
            res.push_back(i);
    }

    return res;
}

int main()
{

    vector<int> nums = {4, 3, 2, 7, 7, 2, 3, 1};
    vector<int> result = find_disappeared_numbers(nums);
    for (auto n : result)
        cout << n << " ";
    return 0;
}